plugins/modules/mso_backup.py

```diff
@@ -0,0 +1,200 @@
+  cisco.mso.mso_backup:
+    host: mso_host
+    username: admin
+    password: SomeSecretPassword
+    backup: Backup
+    state: absent
+  delegate_to: localhost
+
+- name: Query a backup
+  cisco.mso.mso_backup:
+    host: mso_host
+    username: admin
+    password: SomeSecretPassword
+    backup: Backup
+    state: query
+  delegate_to: localhost
+  register: query_result
+
+- name: Query all backups
+  cisco.mso.mso_backup:
+    host: mso_host
+    username: admin
+    password: SomeSecretPassword
+    state: query
+  delegate_to: localhost
+  register: query_result
+'''
+
+RETURN = r'''
+'''
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible_collections.cisco.mso.plugins.module_utils.mso import MSOModule, mso_argument_spec
+
+
+def main():
+    argument_spec = mso_argument_spec()
+    argument_spec.update(
+        location_type=dict(type='str', default='local', choices=['local', 'remote']),
+        description=dict(type='str'),
+        backup=dict(type='str', aliases=['name']),
+        remote_location_name=dict(type='str'),
+        backup_remote_path=dict(type='str'),
+        state=dict(type='str', default='present', choices=['absent', 'present', 'query']),
+    )
+
+    module = AnsibleModule(
+        argument_spec=argument_spec,
+        supports_check_mode=True,
+    )
+
+    description = module.params.get('description')
+    location_type = module.params.get('location_type')
+    state = module.params.get('state')
+    backup = module.params.get('backup')
+    remote_location_name = module.params.get('remote_location_name')
+    backup_remote_path = module.params.get('backup_remote_path')
+
+    mso = MSOModule(module)
+
+    if remote_location_name:
+        remote_location_id = (mso.lookup_remote_location(remote_location_name))[0]
+        remote_path = (mso.lookup_remote_location(remote_location_name))[1]
+
+    if backup_remote_path:
+        remote_path += '/' + backup_remote_path
+
+    backup_id = None
+    path = 'backups'
+    mso.existing = mso.query_objs('backups/backupRecords', key='backupRecords')
+    if backup:
+        if mso.existing:
+            data = mso.existing
+            mso.existing = []
+            for backup_info in data:
+                backup_id = backup_info.get('id')
+                if backup == backup_info.get('name').split('_')[0]:
+                    path = 'backups/backupRecords/{id}'.format(id=backup_id)
+                    mso.existing.append(backup_info)
+            if len(mso.existing) < 1:
+                mso.existing = mso.get_obj(path, name=backup)
+
+    if state == 'query':
+        mso.exit_json()
+
+    if state == 'absent':
+        mso.previous = mso.existing
+        if mso.existing:
+            if module.check_mode:
+                mso.existing = {}
+            else:
+                mso.existing = mso.request(path, method='DELETE')
+        mso.exit_json()
+
+    path = 'backups'
+    mso.existing = mso.get_obj(path, name=backup)
```

Comment:
We can change it to mso.existing = {} 
We need this to be able to do a clean POST operation every time because despite having the same name for backup, we create a new backup from scratch. We wouldn't need it if we had PUT for the backup to be changed.